pkg/cmd/pulumi/package.go

```diff
@@ -216,21 +214,21 @@ func providerFromSource(packageSource string) (plugin.Provider, error) {
 	}
 
 	// We were given a path to a binary or folder, so invoke that.
-	info, err := os.Stat(pkg)
+	info, err := os.Stat(packageSource)
```

Comment:
This is a deliberate change. The existing code previously modified `pkg`, and it felt wrong to modify `descriptor.Name` here (and perhaps leaves us open to mutation causing spooky action at a distance) so I've opted to modify the string from which `descriptor.Name` was originally sourced.